Formula/p/porter.rb

```diff
@@ -0,0 +1,24 @@
+class Porter < Formula
+  desc "Porter enables you to package your application artifact, client tools, configuration and deployment logic together as an installer that you can distribute, and install with a single command."
+  homepage "https://porter.sh"
+  url "https://github.com/SpiffyEight77/porter/archive/refs/tags/v1.0.17.1.tar.gz"
+  sha256 "2139036410612d28a455575066101aa9fcc0e231048d5b7b5c295ba207efd0ff"
+  license "Apache-2.0"
+  head "https://github.com/getporter/porter.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "run", "mage.go", "-v", "Build"
+
+    bin.install "bin/porter"
+
+    generate_completions_from_executable(bin/"porter", "completion")
+  end
+
+  test do
+    assert_predicate bin/"porter", :exist?, "porter binary doesn't exist"
+
+    assert_match "porter v1.0.17.1 (4157be0)", shell_output("#{bin}/porter version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.